app/models/competition.rb

```diff
@@ -96,6 +96,8 @@ class Competition < ApplicationRecord
     restricted: 2,
   }, prefix: true
 
+  enum :registration_version, { v1: 'v1', v2: 'v2', v3: 'v3' }
```

Comment:
In "simple" cases like this, you can use `enum :my_column, [:yes, :no, :maybe]` instead. The database level will be an integer corresponding to array indices, but if you access `some_model.my_column` Rails will convert and return it to you as `:yes`.

Reply:
Independent remark: You can add the option `prefix: true` to this. Any enum in Rails generates magic methods that are based on the enum's entries: An entity with `enum :my_field, [:yes, :no]` will automatically get methods like `my_entity.yes?` and `my_entity.no?`. By adding the `prefix` option, the name of the enum is prefixed like `my_entity.my_field_yes?`.
In this case, I think `my_comp.registration_system_v2?` is much more concise (and much less misleading!) than `my_comp.v2?`

Reply:
Yet another independent remark: You can pass `default: :v2` as an option to the enum call. I think it's self-explanatory what that does.

Reply:
done, but should we not set it explicitly in the form instead of setting a default?

Reply:
No, the competition has to be initialized with "some value" before it's being serialized and passed to the form.
On database level, zero (which means `v1`) should be the default, just in case. That's our hard backup. On Rails level, you can set V2 as default.
We are already effectively treating the current boolean field's `true` as a default, except that Rails didn't offer a standardized way of distinguishing between database-level and model-level defaults. Now with enums, it does. And I think we should leverage that.